pennylane_lightning/src/util/Macros.hpp

```diff
@@ -13,12 +13,155 @@
 // limitations under the License.
```

Comment:
For a series of macros like this, how do you think we can best guarantee full code-coverage on the generated source for all supported platforms?

Reply:
In fact, it would be nice if we have C++ coverage tests in our CI (which we do not have now). As in wheel builders, we may add C++ coverage tests from PPC/Mac(x64 only)/Windows/Linux.